lib/kramdown/document_sections.rb

```diff
@@ -43,13 +43,11 @@ def sectionable_content
     sections.map.with_index do |section, index|
       next_section = sections[index + 1]
 
-      if next_section.present?
-        Kramdown::ContentSection.new(
-          title: section.title, content: content_between(section.start, next_section.end_of_previous_section)
-        )
-      else
-        Kramdown::ContentSection.new(title: section.title, content: content_between(section.start, END_OF_CONTENT))
-      end
+      section_end = next_section&.end_of_previous_section || END_OF_CONTENT
```

Comment:
Nice refactor 🔥

Reply:
Originally considered a ternary for assignment but @JoshDevHub taught me about the short circuiting, which is cool because I never even considered that pattern with JS since it'd work there too.